onnxruntime/core/mlas/lib/qlgavgpool.cpp

```diff
@@ -392,47 +511,71 @@ MlasQLinearGlobalAveragePoolNhwcSingleBatch(
     const __m128i vi6 = _mm_loadl_epi64((const __m128i*)i6); \
     i6 += 8
 
-#define CALCULATE_ACCUMULATE_VECTORS()                                                                 \
-    __m128i vacc_lo = finish_one_pass ? _mm_loadu_si128((__m128i*)acc) : vbias;                        \
-    __m128i vacc_hi = finish_one_pass ? _mm_loadu_si128(((__m128i*)acc) + 1) : vbias;                  \
-    const __m128i vxi0 = _mm_unpacklo_epi8(vi0, vzero);                                                \
-    const __m128i vxi1 = _mm_unpacklo_epi8(vi1, vzero);                                                \
-    const __m128i vxi2 = _mm_unpacklo_epi8(vi2, vzero);                                                \
-    const __m128i vxi3 = _mm_unpacklo_epi8(vi3, vzero);                                                \
-    const __m128i vxi4 = _mm_unpacklo_epi8(vi4, vzero);                                                \
-    const __m128i vxi5 = _mm_unpacklo_epi8(vi5, vzero);                                                \
-    const __m128i vxi6 = _mm_unpacklo_epi8(vi6, vzero);                                                \
-    const __m128i vsum01 = _mm_add_epi16(vxi0, vxi1);                                                  \
-    const __m128i vsum23 = _mm_add_epi16(vxi2, vxi3);                                                  \
-    const __m128i vsum45 = _mm_add_epi16(vxi4, vxi5);                                                  \
-    const __m128i vsum016 = _mm_add_epi16(vsum01, vxi6);                                               \
-    const __m128i vsum2345 = _mm_add_epi16(vsum23, vsum45);                                            \
-    const __m128i vsum = _mm_add_epi16(vsum016, vsum2345);                                             \
-    vacc_lo = _mm_add_epi32(vacc_lo, _mm_unpacklo_epi16(vsum, vzero));                                 \
-    vacc_hi = _mm_add_epi32(vacc_hi, _mm_unpackhi_epi16(vsum, vzero))
+#define CALCULATE_ACCUMULATE_VECTORS()                                                         \
```

Comment:
random thoughts, maybe we should convert these macros to inline functions in the future?
and using an inline function to switch between _mm_srai_epi16(_mm_unpacklo_epi8())  vs _mm_unpacklo_epi8() we can have shorter code to read :-)